### PR TITLE
Add unsupported qos checks on initialize

### PR DESCRIPTION
**Pull request by ross-desmond**
Signed-off-by: Ross Desmond <44277324+ross-desmond@users.noreply.github.com>

*Issue #, if available:*

*Description of changes:*
Add checks for unsupported QoS.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Review by emersonknapp — APPROVED** [review]
looks straightforward

File: rmw_fastrtps_cpp/src/qos.cpp [hunk 1]
```diff
@@ -157,4 +157,32 @@ get_datawriter_qos(
...
+  const rmw_time_t & time)
+{
+  return time.sec == 0 && time.nsec == 0;
+}
+
+bool
+is_valid_qos(
+  const rmw_qos_profile_t & qos_policies)
+{
+  if (!is_time_default(qos_policies.deadline)) {
```
Comment (nburek) [inline rmw_fastrtps_cpp/src/qos.cpp:172]: There should be an enum for deadline and lifespan that you could check as well. They should be set to NONE by default. Long term it probably makes sense to consider that different from a zero initialized time (which may still be considered invalid). 

I wouldn't consider this a blocker by any means.

**Review by nburek — APPROVED** [review]

File: rmw_fastrtps_cpp/src/qos.cpp (hunk 1 above)
Comment (ross-desmond) [reply rmw_fastrtps_cpp/src/qos.cpp:172]: Hmm, does that mean we want to add one? As these do not currently exist in the RMW layer

Comment (nburek) [reply rmw_fastrtps_cpp/src/qos.cpp:172]: We discussed this offline and we're not going to add the enums to the rmw package for deadline and lifespan at this point.

**Pull request merged**